flink-runtime/src/test/java/org/apache/flink/runtime/operators/coordination/OperatorCoordinatorHolderTest.java

```diff
@@ -601,6 +606,8 @@ protected void step() throws Exception {
     private static final class FutureCompletedAfterSendingEventsCoordinator
             extends CheckpointEventOrderTestBaseCoordinator {
 
+        private final OneShotLatch checkpointCompleted = new OneShotLatch();
+
         @Nullable private CompletableFuture<byte[]> checkpoint;
```

Comment:
Might not be super relevant, but we might save some cycles if we mark this field `volatile` since it is accessed from different threads.

Reply:
Can do that, is also nice "documentation".
I think for correctness it is not relevant, since we don't rely on happen-before visibility here. (And visibility will happen eventually anyways, as in a few microseconds later).